rtos/source/TARGET_CORTEX/mbed_lib.json

```diff
@@ -76,6 +76,9 @@
         "STM32F072RB": {
             "main-thread-stack-size": 3072
         },
+        "TMPM46B": {
+            "main-thread-stack-size": 3072
+        },
```

Comment:
Do we really need to specify this per target?
Or can we set the stack-size depending on the total RAM? (which would be target independent)

Reply:
Checking RAM size is ideal, and we do have `target.mbed_ram_size` for this purpose.

The trouble is, we only have it defined for _a few_ targets. For most other target, the RAM and ROM sizes are hardcoded in the linker script (e.g. scatter file) - only the linker knows the RAM, not the application/library/...

Reply:
Well, 17 targets have it defined: https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json
Okay it's more than a few, but still not far from a good coverage.

Reply:
Agree, thanks for clarifying it
This PR is great anyway as it helps us with some clean-up. More could be done in the future.

Reply:
Personally I'd like to see the sizes defined - it can help us to disable certain Greentea test cases for ultra-constrained targets on which we even struggle to run some very basic tests... @evedon

Reply:
The best would be to generate the linker script (having set per toolchain and that is it) so having the size defined is one way forward there :)